Src/Support/Google.Apis/Requests/BatchRequest.cs

```diff
@@ -179,7 +179,35 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
             var result = await httpClient.PostAsync(new Uri(batchUrl), outerContent, cancellationToken)
                 .ConfigureAwait(false);
 
-            result.EnsureSuccessStatusCode();
+            if (!result.IsSuccessStatusCode)
+            {
+                try
+                {
+                    // Parse the error from the current response.
+                    var error = await service.DeserializeError(result).ConfigureAwait(false);
```

Comment:
Does this method *always* succeed (handling both plaintext and JSON)? I'm very keen that we don't make errors harder if the error itself is unusual.

Reply:
(Or is that precisely why this is in a try/catch.)

Reply:
It may fail, if it's not able to read the content or if it can't parse the content as JSON. I can fine grain the catch more to account for those cases and simply call EnsureSuccessStatusCode.